base/array.jl

```diff
@@ -863,22 +863,9 @@ function indexin(a::AbstractArray, b::AbstractArray)
     [get(bdict, i, 0) for i in a]
 end
 
-# findin (the index of intersection)
-function findin(a, b::UnitRange)
-    ind = Array(Int, 0)
-    f = first(b)
-    l = last(b)
-    for i = 1:length(a)
-        if f <= a[i] <= l
-            push!(ind, i)
-        end
-    end
-    ind
-end
-
 function findin(a, b)
     ind = Array(Int, 0)
-    bset = Set(b)
+    bset = unique(b)
     @inbounds for i = 1:length(a)
         a[i] in bset && push!(ind, i)
```

Comment:
>   (in costs O(log k) for k elements int the set)

pretty sure it's O(1).

> (getting the unique elements and testing membership) are separate optimizations

sure. In some sense what we want here is "best 'in' container", which for a set/dict/range will be itself, for small arrays will be it uniqued, and Set of the array for large arrays. I can see that unique pretty much does that (with the exception of Set of a large array - but like I say you can pass that explicitly).

Unique is very cheap here, the cost is at most the worst case find in one of the `a` rows (O(k)), and can be O(1) for many datatypes, so I think it's the right choice.